webui/src/lib/components/repository/ObjectsDiff.jsx

```diff
@@ -0,0 +1,167 @@
+        : <StatDiff left={left.response} right={right.response} diffType={"changed"}/>;
+}
+
+const AddedDiff =  ({repoId, rightRef, path}) => {
+    let readable = readableObject(path);
+    const getFn = readable ? objects.getWithSizeConstraint : objects.getStat;
+    const right = useAPI(async () => getFn(repoId, rightRef, path, maxDiffSizeBytes),
+            [repoId, rightRef, path]);
+
+    if (right.loading) return <Loading/>;
+    if (!!right.error) return <Error error={right.error}/>;
+
+    return readable
+        ? <ContentDiff right={right.response[0]} rightSize={(right.response[1]).get("Content-length")} diffType={"added"}/>
+        : <StatDiff right={right.response} diffType={"added"}/>;
+}
+
+const RemovedDiff =  ({repoId, leftRef, path}) => {
+    let readable = readableObject(path);
+    const getFn = readable ? objects.getWithSizeConstraint : objects.getStat;
+    const left = useAPI(async () => getFn(repoId, leftRef, path, maxDiffSizeBytes),
+            [repoId, leftRef, path]);
+
+    if (left.loading) return <Loading/>;
+    if (!!left.error) return <Error error={left.error}/>;
+
+    return readable
+        ? <ContentDiff left={left.response[0]} leftSize={(left.response[1]).get("Content-length")} diffType={"removed"}/>
+        : <StatDiff left={left.response} diffType={"removed"}/>;
+}
+
+const ContentDiff = ({left, right, leftSize, rightSize, diffType}) => {
+    const newStyles = {
+        variables: {
+            light: {
+                codeFoldGutterBackground: "#6F767E",
+                codeFoldBackground: "#E2E4E5"
+            }
+        }
+    };
+
+    return <div>
+        <span><StatsReportBlock leftSize={leftSize} rightSize={rightSize} diffType={diffType}/></span>
+        <ReactDiffViewer
+            oldValue={left}
+            newValue={right}
+            splitView={false}
+            styles={newStyles}
+            compareMethod={DiffMethod.LINES}/>
+    </div>;
+}
+
+
+const StatDiff = ({left, right, diffType}) => {
+    let statsReportBlock;
+    if (diffType === 'changed') {
+        if (typeof left === 'undefined' || typeof right === 'undefined') {
+            return <Error error={"Invalid input"}/>;
+        }
+        statsReportBlock = <StatsReportBlock leftSize={left.size_bytes} rightSize={right.size_bytes} diffType={diffType}/>;
+    } else if (diffType === 'added') {
+        if (typeof right === 'undefined') {
+            return <Error error={"Invalid input: right is undefined"}/>;
+        }
+        statsReportBlock = <StatsReportBlock rightSize={right.size_bytes} diffType={diffType}/>;
+    } else {
+        if (typeof left === 'undefined') {
+            return <Error error={"Invalid input: left is undefined"}/>;
+        }
+        statsReportBlock = <StatsReportBlock leftSize={left.size_bytes} diffType={diffType}/>;
+    }
+
+    return <>
+        <div className={"stats-diff-block"}>
+            {statsReportBlock}
+        </div>
+    </>;
```

Comment:
something like that can make it more readable - we can pass a property when it is undefined as long as the component addresses it when needed.
```suggestion
     switch (diffType) {
        case 'changed':
            if (!left && !right) return <Error error={"Invalid input"}/>;
        case 'added':
            if (!left) return <Error error={"Invalid input: left is missing"}/>;
        case 'removed':
            if (!right) return <Error error={"Invalid input: right is missing"}/>;
        default:
            return <Error error={"Unknown diff type: " + diffType}/>;
    }
    const rightSize = right && right.size_bytes;
    const leftSize = left && left.size_bytes;
    return <>
            <div className={"stats-diff-block"}>
                <StatsReportBlock leftSize={leftSize} rightSize={rightSize} diffType={diffType}/>
            </div>
           </>;
```